code/modules/atmospherics/machinery/components/unary_devices/cryo.dm

```diff
@@ -87,17 +91,72 @@
 		beaker = null
 
 /obj/machinery/atmospherics/components/unary/cryo_cell/update_icon()
-	if(panel_open)
-		icon_state = "cell-o"
-	else if(state_open)
-		icon_state = "cell-open"
-	else if(on && is_operational())
-		if(occupant)
-			icon_state = "cell-occupied"
+	cut_overlays()
+
+	if(state_open)
+		icon_state = "pod-open"
+	else if(occupant)
+		var/mutable_appearance/occupant_overlay
+
+		if(istype(occupant, /mob/living/carbon/human) || istype(occupant, /mob/living/carbon/alien/larva) || istype(occupant, /mob/living/simple_animal/slime) || istype(occupant, /mob/living/simple_animal/parrot) || istype(occupant, /mob/living/simple_animal/pet))
+			// Mobs that are smaller than cryotube
+			occupant_overlay = mutable_appearance(occupant.icon, occupant.icon_state)
+			occupant_overlay.copy_overlays(occupant)
+
+		else if(istype(occupant, /mob/living/carbon/monkey)) // Monkey
+			occupant_overlay = mutable_appearance(CRYOMOBS, "monkey")
+			occupant_overlay.copy_overlays(occupant)
+
+		else if(istype(occupant, /mob/living/carbon/alien))
+
+			if(istype(occupant, /mob/living/carbon/alien/humanoid/royal)) // Queen and prae
+				occupant_overlay = mutable_appearance(CRYOMOBS, "alienq")
+
+			else if(istype(occupant, /mob/living/carbon/alien/humanoid/hunter)) // Hunter
+				occupant_overlay = mutable_appearance(CRYOMOBS, "alienh")

```

Comment:
I'd like to note that I _really_ don't like this as a solution and would kind of prefer that you crop the mutable appearance to fit the tube, even though that might look weird sometimes.

Maybe also shift large icons like alien queens/praetorians down a little.